collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadListActivity.java

```diff
@@ -321,11 +318,8 @@ private void downloadFormList() {
             }
         } else {
             viewModel.clearFormDetailsByFormId();
-            if (progressDialog != null) {
-                // This is needed because onPrepareDialog() is broken in 1.6.
-                progressDialog.setMessage(viewModel.getProgressDialogMsg());
-            }
-            createProgressDialog();
+            refreshFormListDialogFragment = new RefreshFormListDialogFragment();
+            refreshFormListDialogFragment.show(getSupportFragmentManager(), ProgressDialogFragment.COLLECT_PROGRESS_DIALOG_TAG);
```

Comment:
Like with dismissing you can use `DialogUtils.showIfNotShowing`. Sorry I should have pointed those out before leading you down the `ProgressDialogFragment` path!

Reply:
That's okay! Actually I have used `DialogUtils.showIfNotShowing` before :disappointed: